arch/arm/src/tlsr82/tlsr82_adc.c

```diff
@@ -545,85 +772,79 @@ static uint16_t adc_read(void)
  *
  ****************************************************************************/
 
-#if 0
-
-static void adc_calibrate(void)
+#ifdef CONFIG_TLSR82_ADC_CALI
+static void tlsr82_adc_calibrate(struct adc_chan_s *priv)
 {
-  uint16_t cali_val;
-  uint16_t adc;
-  uint16_t adc_max = 0;
-  uint16_t adc_min = UINT16_MAX;
-  uint32_t adc_sum = 0;
-  uint32_t regval;
+  uint8_t cali_data[7];
 
-  regval = read_efuse(ADC_CAL_BASE_REG, ADC_CAL_VER_OFF, ADC_CAL_VER_LEN);
-  if (regval == 1)
-    {
-      ainfo("Calibrate based on efuse data\n");
-
-      regval = read_efuse(ADC_CAL_BASE_REG, ADC_CAL_DATA_OFF,
-                          ADC_CAL_DATA_LEN);
-      cali_val = regval + ADC_CAL_DATA_COMP;
-    }
-  else
-    {
-      ainfo("Calibrate based on GND voltage\n");
+  uint32_t base_vref;
+  uint32_t vbat_vref;
 
-      /* Enable Vdef */
+  (void)memset((void *)cali_data, 0, 7);
```

Comment:
```suggestion
  memset(cali_data, 0, sizeof(cali_data));
```
I think we do not enable "unused result" warning. Or am I wrong?

Reply:
Ack and done.